File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ProtocolCodeLens.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ProtocolCodeLens(
  val range: Range,
  val command: ProtocolCommand? = null,
  val isResolved: Boolean,
)

File: agent/bindings/csharp/ProtocolCodeLens.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolCodeLens
  {
    [JsonProperty(PropertyName = "range")]
    public Range Range { get; set; }
    [JsonProperty(PropertyName = "command")]
    public ProtocolCommand Command { get; set; }
    [JsonProperty(PropertyName = "isResolved")]
    public bool IsResolved { get; set; }
  }
}
